benchmarks/jmh/README.md

```diff
@@ -42,33 +33,35 @@ $ ./gradlew --no-daemon :benchmarks:jmh:clean :benchmarks:jmh:jmh ...
 
 ## Retrieving flame graph using async-profiler
 
-Allow running `perf` as a normal user:
-
-```
-# echo 1 > /proc/sys/kernel/perf_event_paranoid
-# echo 0 > /proc/sys/kernel/kptr_restrict
-```
-
-Install [async-profiler](https://github.com/jvm-profiling-tools/async-profiler) and
-[FlameGraph](https://github.com/brendangregg/FlameGraph):
-
-```
-$ cd "$HOME"
-$ git clone https://github.com/jvm-profiling-tools/async-profiler.git
-$ git clone https://github.com/brendangregg/FlameGraph.git
-$ cd async-profiler
-$ make
-```
+- Allow running `perf` as a normal user on Linux:
+  ```
+  # echo 1 > /proc/sys/kernel/perf_event_paranoid
+  # echo 0 > /proc/sys/kernel/kptr_restrict
+  ```
+  - MacOS profiling is limited to user space code only, thus this does not work with MacOS.
 
-When running a benchmark, specify `-Pjmh.profilers` option:
+- Install [async-profiler](https://github.com/jvm-profiling-tools/async-profiler):
+  ```
+  $ cd "$HOME"
+  $ git clone https://github.com/jvm-profiling-tools/async-profiler.git
+  $ cd async-profiler
+  $ make
+  ```
 
-```
-$ ./gradlew --no-daemon :benchmarks:jmh:clean :benchmarks:jmh:jmh \
-  "-Pjmh.profilers=jmh.extras.Async:asyncProfilerDir=$HOME/async-profiler;flameGraphDir=$HOME/FlameGraph"
-```
+- When running a benchmark, specify `-Pjmh.profilers` option:
+  - On Linux
+    ```
+    $ ./gradlew :benchmarks:jmh:jmh \
+      "-Pjmh.profilers=async:libPath=$HOME/async-profiler/build/lib/libasyncProfiler.so;output=flamegraph;dir=$HOME/result"
+    ```
+  - On MacOS
+    ```
+    $ ./gradlew :benchmarks:jmh:jmh \
+      "-Pjmh.profilers=async:libPath=$HOME/async-profiler/build/lib/libasyncProfiler.dylib;output=flamegraph;dir=$HOME/result"
+    ```
 
 ## Notes
 
 - Do not forget to wrap `-Pjmh.params` and `-Pjmh.profilers` option with double quotes, because otherwise your
   shell will interpret `;` as the end of the command.
-- See [sbt-jmh documentation](https://github.com/ktoso/sbt-jmh#using-async-profiler) for more profiler options.
+- Run `$ ./gradlew :benchmarks:jmh:jmh "-Pjmh.profilers=async"` for more profiler options.
```

Comment:
Running this actually threw an exception for me 😅 
```bash
Profilers failed to initialize, exiting.
Unable to load async-profiler. Ensure asyncProfiler library is on LD_LIBRARY_PATH (Linux), DYLD_LIBRARY_PATH (Mac OS), or -Djava.library.path. Alternatively, point to explicit library location with -prof async:libPath=<path>.
```

Reply:
Oops, fixed, thanks! 😓